### ISSUE TITLE
Setting "visible message actions" does not seem to have any effect

### ISSUE THREAD

**Issue by Garfonso** [thunderbird/thunderbird-android#5448]
**Describe the bug**
In settings there is an option that lets you select "visible message actions". In the past this did determine which buttons were visible in message view. Today those settings do not do anything... the buttons in the message action thing do not change at all (and there is not enough space there for more than one button... so that is completely broken).

**To Reproduce**
1. Go to 'settings -> general settings -> display -> press visible message actions', select a few
2. Open a message, see that there is just one button and it does not reflect what you had selected.

**Expected behavior**
The buttons in the action bar should be what I select in this option.

**Screenshots**
![Screenshot_20210726_093210_com fsck k9](https://user-images.githubusercontent.com/620860/126950689-aec5517b-f878-4ec0-8007-1c0b8ff168a4.jpg)
![Screenshot_20210726_093227_com fsck k9](https://user-images.githubusercontent.com/620860/126950692-563ee429-4201-4383-9970-a94caa3545dc.jpg)

**Environment (please complete the following information):**
 - K-9 Mail version: 5.8
 - Android version: 10
 - Device: Huawei Mate 20 Pro
 - Account type: IMAP

**Comment by MirLach** [comment]
I tried Thunderbird 11.1 but I don't like the Unified Inbox interface with very strange UI for selecting another account so I tried **Thunderbird Beta 12.0b7** which I like more, but there is no button or menu item for marking message as `Spam` (and moving to spam folder). I went to `Settings / General settings / Display / Visible message actions`   and checked boxes for `Delete` and `Spam`. But there is no `Spam` button in the message list view, nor in the message view, nor in the menu (even not under `Refile` submenu). No matter what is selected in the `Visible message actions`, nothing has changed with buttons or tree dot menu items in message list and message view. 
 `Visible message actions` works in 11.1 but doesn't work in 12.0b7 so it is some kind or regression.

**Comment by coreycb** [comment]
@MirLach This works for me on 12.0b7. When I open a message, I can see the various actions.

**Comment by MirLach** [comment]
Can you see it even in message list view? I cannot see any actions beside delete in list view or message detail view, nor in the menu.

<img width="864" height="1062" alt="Image" src="https://github.com/user-attachments/assets/247a705f-4df1-4320-b1d7-92a77a40c0da" />

<img width="864" height="688" alt="Image" src="https://github.com/user-attachments/assets/6e7073c3-a62d-4016-b189-ffe486d9e4f0" />

<img width="864" height="688" alt="Image" src="https://github.com/user-attachments/assets/a9970a1c-57cc-49b2-90ae-b3461d3bacb0" />

<img width="864" height="688" alt="Image" src="https://github.com/user-attachments/assets/e156a018-baf0-44af-9164-9ac2f2f14047" />

**Comment by MirLach** [comment]
I upgraded Thunderbird to 12.1 and now this production release doesn't have the mentioned `Spam` button too. So all current versions of Thunderbird are broken for me and I cannot use it anymore. Tested on Pixel 8 with GrapheneOS based on Android 16. No matter what I select in "`Visible message actions`", `Spam` button / menu item disappeared.

**Comment by platform34** [comment]
Have you checked the spam and other special folders are mapped correctly for each account. I.e settings > account (eg gmail) > folders.
The spam menu option does not show when the spam folder is set to none.
I'm using TB 12.1 & 13.0b1 and the spam option is visible.

**Comment by MirLach** [comment]
Yes, this was the problem. Somehow Thunderbird lost the relation to all subfolders like Spam, Trash, Drafts and Sent after the upgrade.
When I set these relations manually again, I have `Spam` item in the menu. But there is another problem, that all folders above (beside Inbox) are empty even after I pull down to "**Load up to 500 more**" and when I manually mark some message as Spam, I can see it in Thunderbird mobile app, but not anywhere else - it is not visible on the server, so all these folders works only locally.

**Comment by platform34** [comment]
Your second issue sounds like #9567 . I'd recommend trying the [workaround](https://github.com/thunderbird/thunderbird-android/issues/9567#issuecomment-3183117087) within.

Disable "Auto-detect Imap namespace" under Settings > account > fetching mail > incoming server and leave "Imap path prefix" empty.

**Comment by k5e** [comment]
I have the original issue where the "visible actions" don't show on K9 Mail 12.1.

The UI changed in so many ways recently that I did not notice the missing "mark as Spam" button immediately.

I miss it.
